Source/GUI/MainWindow.cpp

```diff
@@ -15,12 +16,22 @@
 
 MainWindow::MainWindow()
 {
+  setWindowTitle("Dolphin Memory Engine");
   initialiseWidgets();
   makeLayouts();
   makeMenus();
   DolphinComm::DolphinAccessor::init();
   makeMemViewer();
   firstHookAttempt();
+
+  if (SConfig::getInstance().getMainWindowGeometry().size())
+    restoreGeometry(SConfig::getInstance().getMainWindowGeometry());
+  if (SConfig::getInstance().getMainWindowState().size())
+    restoreState(SConfig::getInstance().getMainWindowState());
+
+  m_watcher->restoreWatchModel(SConfig::getInstance().getWatchModel());
+
+  hookIfNotHooked();
```

Comment:
A few lines above this the firstHookAttempt is already called. Is there a particular reason we should keep this?

Reply:
At the moment, `hookIfNotHooked()` schedules _itself_ for another run in 3000 ms, so it needs this first call to start the loop.

This call could be changed to
```cpp
QTimer::singleShot(3000, this, &MainWindow::hookIfNotHooked);
```
, so the loop is started without trying to hook.

I think my preference would be to remove the `firstHookAttempt()` function, and add the `updateMem2Status()` call directly in `hookIfNotHooked()`, and then we can keep this call to `hookIfNotHooked()` here, which then starts the cycle.

Reply:
This has been done.

(I finally remembered why `updateMem2Status()` was not being called in `hookIfNotHooked()`: it's already called in `onHookAttempt()`.)